### PR TITLE
I18n/rbac cm manager attempt fix

### PR DESCRIPTION
**Pull request by mfrachet**
### What does it do?

- Adds a middleware inside i18n that will catch the `ContentManager/RBACManager/SET_PERMISSIONS` actions
- The previously mentioned middleware modifies the permission in the array and check if the actual locale is allowed

### CODE REVIEW

File: packages/strapi-plugin-content-manager/admin/src/containers/ListView/ListViewLayout.js [hunk 1]
```diff
@@ -19,18 +20,21 @@ const ListViewLayout = ({ layout, ...props }) => {
     if (initialParams) {
       setQuery(initialParams);
     }
+
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [initialParams]);
 
-  if (!initialParams) {
+  useEffect(() => {
```
Comment (mfrachet) [inline packages/strapi-plugin-content-manager/admin/src/containers/ListView/ListViewLayout.js:27]: ℹ️ extracting the cleanup from the listview here

File: packages/strapi-plugin-i18n/admin/src/middlewares/localePermissionMiddleware.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import get from 'lodash/get';
+
+const filterPermissionWithLocale = locale => permission =>
```
Comment (mfrachet) [inline packages/strapi-plugin-i18n/admin/src/middlewares/localePermissionMiddleware.js:3]: This is problematic for the super admin I think no? 🤔

File: packages/strapi-plugin-content-manager/admin/src/containers/ListView/ListViewLayout.js [hunk 3]
```diff
@@ -2,14 +2,15 @@ import React, { useEffect } from 'react';
...
+import { resetProps, setLayout } from './actions';
 import ListView from './index';
+import useSyncRbac from '../RBACManager/useSyncRbac';
 
 const ListViewLayout = ({ layout, ...props }) => {
   const dispatch = useDispatch();
   const initialParams = useSelector(state => state.get('content-manager_listView').initialParams);
-  const [, setQuery] = useQueryParams(initialParams);
+  const [{ query }, setQuery] = useQueryParams(initialParams);
+  const permissions = useSyncRbac(query, props.slug);
```
Comment (mfrachet) [inline packages/strapi-plugin-content-manager/admin/src/containers/ListView/ListViewLayout.js:13]: When dealing with the ListView in the CM, it deals with a lot a things. In fact, many things are actually coupled to the ListView lifecycle.

Now, with i18n, we need to destroy/re-create the listview based on the locale that exists in the query string.

In order to do so, the permission check has been moved higher in the React tree. Now, the ListViewLayout is the one responsible for dealing with the permissions and the ListView only consumes them.

File: packages/strapi-plugin-content-manager/admin/src/containers/RBACManager/useSyncRbac.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import { useDispatch, useSelector } from 'react-redux';
+import pluginId from '../../pluginId';
+import { resetPermissions, setPermissions } from './actions';
+
+const selectPermissions = state => state.get(`${pluginId}_rbacManager`).permissions;
+
+const selectCollectionTypePermissions = state =>
+  state.get('permissionsManager').collectionTypesRelatedPermissions;
+
+const useSyncRbac = (query, collectionTypeUID = 'listView') => {
```
Comment (mfrachet) [inline packages/strapi-plugin-content-manager/admin/src/containers/RBACManager/useSyncRbac.js:11]: ℹ️ This is basically the RBACManager in a hook fashion

**Review by HichamELBSI — APPROVED** [review]
LGTMerge, good job 👍🏽

File: packages/strapi-plugin-content-manager/admin/src/containers/RBACManager/useSyncRbac.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  useEffect(() => {
+    if (query && query.pluginOptions && relatedPermissions) {
+      dispatch(setPermissions(relatedPermissions, query.pluginOptions, 'listView'));
+
+      return () => {
+        dispatch(resetPermissions());
+      };
+    }
+
+    return () => {};
```
[the remark this answers is not in the document]
Comment (mfrachet) [reply packages/strapi-plugin-content-manager/admin/src/containers/RBACManager/useSyncRbac.js:27]: eslint 😓

File: packages/strapi-plugin-content-manager/admin/src/containers/RBACManager/useSyncRbac.js (hunk 4 above)
Comment (soupette) [reply packages/strapi-plugin-content-manager/admin/src/containers/RBACManager/useSyncRbac.js:11]: Why do you set the collectionTypeUID to listview I don't think it's needed.

File: packages/strapi-plugin-i18n/admin/src/components/LocalePicker/index.js [hunk 6]
```diff
@@ -34,16 +34,30 @@ const EllipsisParagraph = styled(Text)`
   text-align: left;
 `;
 
+const getInitialLocale = (query, locales = []) => {
+  const localeFromQuery = get(query, 'query.pluginOptions.locale', undefined);
+
+  if (localeFromQuery) {
```
[the remark this answers is not in the document]
Comment (mfrachet) [reply packages/strapi-plugin-i18n/admin/src/components/LocalePicker/index.js:40]: Do you think it's necessary for now? Not sure what is the pros of switching to this (except changing the syntax)?

**Pull request merged**